src/styles/main.css

```diff
@@ -1,76 +1,229 @@
+
+    --background: var(--gray-800);
+    --background-highlight: var(--gray-900);
+
+    --disabled: var(--gray-500);
+
+    /* ! Deprecating neutral */
+    --neutral: var(--gray-900);
+
+    /* Misc sematics: light mode */
+    --tooltipShadow: rgba(255, 255, 255, 0.24);
+    --switchBackground: rgba(255, 255, 255, 0.24);
+    --hubHeroContentBg: rgba(34, 34, 34, 0.8);
+    --bg-main-gradient: linear-gradient(
+      102.7deg,
+      rgba(185, 185, 241, 0.2) 0%,
+      rgba(84, 132, 234, 0.2) 51.56%,
+      rgba(58, 142, 137, 0.2) 100%
+    );
+    --table-box-shadow: 0 14px 66px hsla(0, 0%, 96.1%, 0.07),
+      0 10px 17px hsla(0, 0%, 96.1%, 0.03), 0 4px 7px hsla(0, 0%, 96.1%, 0.05);
+    --table-item-box-shadow: 0 1px 1px hsla(0, 0%, 100%, 0.1);
+  }
+}
 
-      --accent: 0 0% 96.1%;
-      --accent-foreground: 0 0% 9%;
+@layer base {
+  body {
+    @apply bg-background font-body leading-base text-body;
+  }
 
-      --destructive: 0 84.2% 60.2%;
-      --destructive-foreground: 0 0% 98%;
+  a {
+    @apply text-primary underline;
+  }
 
-      --border: 0 0% 89.8%;
-      --input: 0 0% 89.8%;
-      --ring: 0 0% 3.9%;
+  /* Should be replaced with list component styles */
+  ul,
+  ol {
+    margin: 0px 0px 1.45rem 1.45rem;
+    padding: 0;
 
-      --radius: 0.5rem;
+    & li {
+      padding-inline-start: 0;
     }
+  }
 
-    .dark {
-      --background: 0 0% 3.9%;
-      --foreground: 0 0% 98%;
+  li {
+    margin-bottom: calc(1.45rem / 2);
 
-      --card: 0 0% 3.9%;
-      --card-foreground: 0 0% 98%;
+    & > ol,
+    & > ul {
+      margin-inline-start: 1.45rem;
+      margin-block: calc(1.45rem / 2);
+    }
 
-      --popover: 0 0% 3.9%;
-      --popover-foreground: 0 0% 98%;
+    & * {
+      @apply last:mb-0;
+    }
 
-      --primary: 0 0% 98%;
-      --primary-foreground: 0 0% 9%;
+    & > p {
+      margin-bottom: calc(1.45rem / 2);
+    }
+  }
 
-      --secondary: 0 0% 14.9%;
-      --secondary-foreground: 0 0% 98%;
+  /* Anchor tag styles */
+  /* Selected specifically for mdx rendered side icon link */
+  .header-anchor {
+    position: "relative !important";
+    display: "initial";
+    margin-inline-start: "-1.5em";
+    padding-inline-end: "0.5rem !important";
+    font-size: "1rem";
+    vertical-align: "middle";
+
+    & svg {
+      @apply invisible inline fill-primary;
+    }
 
-      --muted: 0 0% 14.9%;
-      --muted-foreground: 0 0% 63.9%;
+    &:focus svg {
+      @apply visible;
+    }
+  }
 
-      --accent: 0 0% 14.9%;
-      --accent-foreground: 0 0% 98%;
+  h1:hover,
+  h2:hover,
+  h3:hover,
+  h4:hover,
+  h5:hover,
+  h6:hover {
+    & .header-anchor svg {
+      @apply visible;
+    }
+  }
 
-      --destructive: 0 62.8% 30.6%;
-      --destructive-foreground: 0 0% 98%;
+  pre,
+  code,
+  kbd,
+  samp {
+    @apply font-monospace text-base leading-base;
+  }
+}
 
-      --border: 0 0% 14.9%;
-      --input: 0 0% 14.9%;
-      --ring: 0 0% 83.1%;
-    }
+@layer components {
+  .accordion-trigger {
```

Comment:
Either dump all the utility classes used for a primitive part in here with a single class name, or have the utilities directly declared in the component file.

Though what to do about readability here with components that have variants? See `tailwind/ui/button.tsx`

Reply:
Now that I removed the button component here, take a look at the [source file](https://github.com/shadcn-ui/ui/blob/main/apps%2Fwww%2Fregistry%2Fdefault%2Fui%2Fbutton.tsx)

Reply:
@TylerAPfledderer didn't get why we are doing this approach. Could you elaborate more?

Don't understand why we are not using the same form as in Shadcn: just using the utility classes on the component. I think it is much clearer and easier to tweak, right?

Reply:
I was just presenting this approach to you for inquiry of opinion.

If we have all the utilities under a class like I have here, then all those utilities will not render separately when viewing dev tools, so the styling is easier to read and digest because all the styling would still be listed under that one class name.

The downside here is that we are separating general styling from the way we would create variants, such as the approach ShadCN would take in the button component.

I don't have a problem either way and can personally manage the readability if all the styling was kept directly at the component. The prettier plugin should also provide the ability to sort the order of the utility classes used.

Reply:
Oh I see now. Ok, that is interesting to know.

I would say that, at least for this initial stage, lets keep with the "tailwind/shadcn" convention since all the documentation/examples are pointing in that direction (I think it will be clearer for most devs and contributors) and because I see those downsides that you mention as important. IMO the DX having this separated files is going to be worse than using the classes on the component itself where you can easily find and modify them.

The other minor (but important to us at this point) is performance. Defining new classes will increase the size of the generated css as you are not reusing the utility classes on the compiled bundles.

Reply:
I'll merge this PR as it is and we can tweak these things later.